### DISCUSSION TITLE
Can useInfiniteQuery handle dependencies?

### DISCUSSION THREAD

**Discussion by tg44** [TanStack/query#3725] [Q&A] [answered]
If the client which gives us back the data comes from a hook, can we use that client as a "dependency"?
```ts
  const supabase = useSupabase(serverId);
  const { data, status, fetchNextPage, hasNextPage } = useInfiniteQuery(
    'infiniteMessages',
    async ({ pageParam = 0 }) => {
      console.log('startLoad');
      if (supabase) { // <- here supabase come from a hook and is a `null | Promise<SupabaseClient>`
        const sb = await supabase;
        return await loadMessages(sb, channelId, pageParam);
      } else {
        console.log('noSupabase'); // <- sometimes this happenes
        return [] as Message[];
      }
    },
    {
      getNextPageParam: (lastPage, pages) => {
        if (lastPage.length === loadAtOnce) {
          return pages.length * loadAtOnce;
        } else {
          return null;
        }
      },
    }
  );
```

`useInfiniteQuery` should get an array like `useEffect`, or `useCallback` for this.

My current method is handling the client init in an upper component, but that is not exactly what I want...

EDIT: handling a client in an upper component is still sporadical, no idea why...

**Comment by TkDodo** [comment, marked as answer]
> useInfiniteQuery should get an array like useEffect, or useCallback for this.

that's what the querykey is for. all dependencies of the queryFn should go to the querykey.
